### PR TITLE
Bug 1890141: move off docker.io in templates/builds/jenkins tests 

### PR DESCRIPTION
**Pull request by gabemontero**
@openshift/openshift-team-build-api FYI

@sbose78 this PR currently assumes we can push `docker.io/openshift/test-build-simples2i:latest` to `quay.io/redhat-developer/test-build-simples2i:latest`

/assign @adambkaplan

### CODE REVIEW

**Comment by sbose78** [comment]
That was quick, thank you Gabe!

**Comment by gabemontero** [comment]
Other than the aforementioned ` quay.io/redhat-developer/test-build-simples2i:latest` not existing yet, this look OK so far e2e wise.
Or at least that is the vast majority of failures.

@sbose78 @adambkaplan - do either of your have to authority to create such a repo in quay.io/redhat-developer so I can push to it like I did with the recent jenkins e2e PR that merged ?

**Comment by gabemontero** [comment]
ok @adambkaplan has created `quay.io/redhat-developer/test-build-simples2i:latest`

/retest

**Comment by gabemontero** [comment]
OK I think we are down to multistage.go failing consistently ... the switch to `quay.io/fedora/fedora:34-x86_64` does not seem to be happy (though it is not an image pull thing).

gonna try `registry.access.redhat.com/rhel7` instead (since this is a from via dockerfile test, not going to do registry.redhat.io)

**Comment by gabemontero** [comment]
OK @adambkaplan we are running clean  (enough - I believe the e2e-cmd errors are non related flakes) with this move off of docker.io images 

PTAL

/retest

**Comment by gabemontero** [comment]
Also @adambkaplan I'm thinking we should open/associate a BZ with this and backport to releases we still actively maintain (and I shudder to say ... 3.11 even)

thoughts?

**Comment by gabemontero** [comment]
again unrelated cmd flakes ... cmds that failed different than last run

/retest

**Comment by gabemontero** [comment]
There *might* be a known `oc` bug related to the recent e2e-cmd errors.  I've reached out to Maciej for opinions.  

Note, this test did pass back with https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/25595/pull-ci-openshift-origin-master-e2e-cmd/1314302332669792256 on Oct 8 .... I'm still of the belief that nothing in this PR would have lead to the subsequent failed runs.

**Review by adambkaplan — COMMENTED** [review]
Generally looks good.

I have a concern about using `registry.redhat.io` - will this present a challenge backporting to 4.4 and 3.11? Do you plan on using `registry.access.redhat.com` for those?

I think moving forward `registry.redhat.io` is the right host for Red Hat images, since `registry.access.redhat.com` is somewhat deprecated.

**Comment by gabemontero** [comment]
> Generally looks good.
> 
> I have a concern about using `registry.redhat.io` - will this present a challenge backporting to 4.4 and 3.11? Do you plan on using `registry.access.redhat.com` for those?
> 
> I think moving forward `registry.redhat.io` is the right host for Red Hat images, since `registry.access.redhat.com` is somewhat deprecated.

exactly ... my general plan is to use the TBR until I can't but of course as a result that will entail manual cherrypicks vs. the bot

**Comment by gabemontero** [comment]
different set of cmd test failures with https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/25595/pull-ci-openshift-origin-master-e2e-cmd/1318905854161850368 

this time just timeout connecting to api server flakes

still unrelated to these changes

/retest

**Comment by gabemontero** [comment]
@adambkaplan we have managed to avoid e2e-cmd flakes ... all tests green ... how about an lgtm

oh, and based on the files touched, also need an approval from 

/assign @bparees 

for https://github.com/openshift/origin/blob/master/test/OWNERS

to complete this move off of docker.io images

File: test/extended/builds/multistage.go [hunk 1]
```diff
@@ -23,7 +23,7 @@ var _ = g.Describe("[sig-builds][Feature:Builds] Multi-stage image builds", func
 		testDockerfile = `
 FROM scratch as test
 USER 1001
-FROM centos:7
+FROM registry.access.redhat.com/rhel7
```
Comment (bparees) [inline test/extended/builds/multistage.go:26]: you're using registry.redhat.io elsewhere, might as well use it here too?

Comment (gabemontero) [reply test/extended/builds/multistage.go]: update pushed ... for some reason I wasn't convinced the cred on the host was always used, but in hindsight I think you're right

let's see how long it takes to get another clean e2e-cmd run :-(

**Review by adambkaplan — APPROVED** [review]
/lgtm

**Pull request merged**